tools/depcheck/pkg/cmd/trace.go

```diff
@@ -0,0 +1,293 @@
+package cmd
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/gonum/graph"
+	"github.com/gonum/graph/concrete"
+	"github.com/gonum/graph/encoding/dot"
+)
+
+var traceImportsExample = `# create a dependency graph
+%[1]s trace --root=pkg/one --root=pkg/two
+
+# create a dependency graph and output in "dot" format
+%[1]s trace --root=pkg/one --output=dot
+`
+
+var (
+	validPackages = []string{
+		"/vendor/bitbucket.org/",
+		"/vendor/cloud.google.com",
+		"/vendor/github.com",
+		"/vendor/go4.org",
+		"/vendor/golang.org",
+		"/vendor/google.golang.org",
+		"/vendor/gopkg.in",
+		"/vendor/k8s.io/",
+		"/vendor/vbom.ml",
+	}
+	validDepPrefixes = []string{"github.com/openshift/origin"}
+)
+
+type TraceImportsOpts struct {
+	BaseDir string
+	Roots   []string
+
+	OutputFormat string
+
+	Out    io.Writer
+	ErrOut io.Writer
+}
+
+type TraceImportsFlags struct {
+	OutputFormat string
+	Roots        []string
+}
+
+func (o *TraceImportsFlags) ToOptions(out, errout io.Writer) (*TraceImportsOpts, error) {
+	return &TraceImportsOpts{
+		Roots: o.Roots,
+
+		OutputFormat: o.OutputFormat,
+
+		Out:    out,
+		ErrOut: errout,
+	}, nil
+}
+
+func NewCmdTraceImports(parent string, out, errout io.Writer) *cobra.Command {
+	flags := &TraceImportsFlags{}
+
+	cmd := &cobra.Command{
+		Use:     "trace --root=pkg/one",
+		Short:   "Creates and analyzes a dependency graph from a set of given root packages",
+		Long:    "Creates and analyzes a dependency graph from a set of given root packages",
+		Example: fmt.Sprintf(traceImportsExample, parent),
+		RunE: func(c *cobra.Command, args []string) error {
+			opts, err := flags.ToOptions(out, errout)
+			if err != nil {
+				return err
+			}
+
+			if err := opts.Complete(); err != nil {
+				return err
+			}
+			if err := opts.Validate(); err != nil {
+				return err
+			}
+			if err := opts.Run(); err != nil {
+				return err
+			}
+
+			return nil
+		},
+	}
+
+	cmd.Flags().StringSliceVar(&flags.Roots, "root", flags.Roots, "set of entrypoints for dependency trees used to generate a depedency graph.")
```

Comment:
You need excludes.  We need to exclude cmd/cluster-capacity and cmd/service-catalog